include/cpp2util.h

```diff
@@ -2084,24 +2087,90 @@ struct args_t
     auto end()    const -> iterator    { return iterator{ argc, argv, argc }; }
     auto cbegin() const -> iterator    { return begin(); }
     auto cend()   const -> iterator    { return end(); }
-    auto size()   const -> std::size_t { return cpp2::unsafe_narrow<std::size_t>(argc); }
+    auto size()   const -> std::size_t { return cpp2::unsafe_narrow<std::size_t>(ssize()); }
     auto ssize()  const -> int         { return argc; }
 
     auto operator[](int i) const {
-        if (0 <= i && i < argc) { return std::string_view{ argv[i] }; }
-        else                    { return std::string_view{}; }
+        if (0 <= i && i < ssize())     { return std::string_view{ argv[i] }; }
+        else                           { return std::string_view{}; }
     }
 
     mutable int        argc = 0;        //  mutable for compatibility with frameworks that take 'int& argc'
     char**             argv = nullptr;
 };
 
-inline auto make_args(int argc, char** argv) -> args_t
+inline auto make_args(int argc, char** argv) -> args
 {
-    return args_t{argc, argv};
+    return args{argc, argv};
 }
 
 
+//-----------------------------------------------------------------------
+//
+//  range: a half-open range of [first, last)
```

Comment:
Thanks! Will fix comment.

I'm reluctant to make this more complicated (e.g., bifurcate into two types, use a policy parameter), because the current design seems to work well with all the code identical except only for one additional test in the constructor only?